### PR TITLE
Update ed25519-zebra to 4.0.0.

### PR DESCRIPTION
**Pull request by nuttycom**
This deduplicates sha2, rand-core, block-buffer, digest, and ahash. (It adds duplications for hashbrown and libm which are less important.)

### CODE REVIEW

**Comment by nuttycom** [comment]
Subsumes #6705, #6708

File: qa/supply-chain/audits.toml [hunk 1]
```diff
@@ -24,6 +24,16 @@ who = "Jack Grigg <jack@electriccoin.co>"
...
+[[audits.allocator-api2]]
+who = "Daira Emma Hopwood <daira@jacaranda.org>"
+criteria = "safe-to-deploy"
+version = "0.2.14"
+notes = """
+This is a workaround for https://doc.rust-lang.org/unstable-book/library-features/allocator-api.html being unstable.
+It copies the unstable implementation, and inherently requires a lot of unsafe code, which was infeasible to review completely.
+However, safety requirements appear to be well-documented.
+"""
+
```
Comment (str4d) [inline qa/supply-chain/audits.toml:36]: I am not comfortable with including this audit, as it does not meet the [`safe-to-deploy` standard](https://mozilla.github.io/cargo-vet/built-in-criteria.html#safe-to-deploy) (by virtue of the unsafe blocks not having been fully reviewed). We might want to introduce a mid-point standard of review (that implies `safe-to-run` and is implied by `safe-to-deploy`) that can be used when we don't have time for a full `unsafe` / powerful capability audit. But for now we should remove this and replace it with an exemption.
```suggestion
```

**Review by str4d — CHANGES_REQUESTED** [review]

**Review by daira — APPROVED** [review]
utACK

**Review by str4d — APPROVED** [review]
utACK 5edacdf68e36fb4249017ddd72b0f135e92b2742

**Pull request merged**